### PR TITLE
Bump go version to 1.22.7

### PR DESCRIPTION
**Pull request by mhutchinson**
Updated lint and vuln checks to pin to the version in go.mod file.

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -10,13 +10,12 @@ jobs:
   golangci:
     name: lint
     runs-on: ubuntu-latest
-    steps:
```
Comment (roger2hk) [inline .github/workflows/golangci-lint.yml:13]: Add this back?

**Review by roger2hk — APPROVED** [review]

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (mhutchinson) [reply .github/workflows/golangci-lint.yml:13]: Done, good spot. Would have expected a failing workflow...

Comment (roger2hk) [reply .github/workflows/golangci-lint.yml:13]: That's a required check, so it is easy to spot this time!

**Review by roger2hk — APPROVED** [review]

**Comment by mhutchinson** [comment]
This is a bit messy. ClusterFuzz isn't enjoying the new X.Y.Z go.mod versions, and wants only X.Y. I'm trying to track down where this is coming from.

Looking in the build logs we see it pulls `gcr.io/oss-fuzz-base/base-builder-go@sha256:c7f01a616db0fc554d98edbd7d49a7177197f77fd4e1371bcb7774d7eac8216d`, which is quite old now:

```shell
docker run -it --entrypoint /bin/sh gcr.io/oss-fuzz-base/base-builder-go@sha256:c7f01a616db0fc554d98edbd7d49a7177197f77fd4e1371bcb7774d7eac8216d
# /root/.go/bin/go version
go version go1.19 linux/amd64
```

There is a newer image for this which bundles with go1.22:

```shell
docker run -it --entrypoint /bin/sh gcr.io/oss-fuzz-base/base-builder-go@sha256:37d560af1d31ae8d339902acaedb8eb6fdf3f04469706f39e3bbea44fa3fdeec
# /root/.go/bin/go version
go version go1.22.1 linux/amd64
```

If we can get this to pull the later container for `base-builder-go` then I expect this will resolve itself.

**Comment by roger2hk** [comment]
> This is a bit messy. ClusterFuzz isn't enjoying the new X.Y.Z go.mod versions, and wants only X.Y. I'm trying to track down where this is coming from.
> 
> Looking in the build logs we see it pulls `gcr.io/oss-fuzz-base/base-builder-go@sha256:c7f01a616db0fc554d98edbd7d49a7177197f77fd4e1371bcb7774d7eac8216d`, which is quite old now:
> 
> ```shell
> docker run -it --entrypoint /bin/sh gcr.io/oss-fuzz-base/base-builder-go@sha256:c7f01a616db0fc554d98edbd7d49a7177197f77fd4e1371bcb7774d7eac8216d
> /root/.go/bin/go version
> go version go1.19 linux/amd64
> ```
> 
> There is a newer image for this which bundles with go1.22:
> 
> ```shell
> docker run -it --entrypoint /bin/sh gcr.io/oss-fuzz-base/base-builder-go@sha256:37d560af1d31ae8d339902acaedb8eb6fdf3f04469706f39e3bbea44fa3fdeec
> /root/.go/bin/go version
> go version go1.22.1 linux/amd64
> ```
> 
> If we can get this to pull the later container for `base-builder-go` then I expect this will resolve itself.

Does the following changes in `go.mod` work before oss-fuzz updates the container image?

```
go 1.22
toolchain 1.22.7
```

**Comment by roger2hk** [comment]
> > This is a bit messy. ClusterFuzz isn't enjoying the new X.Y.Z go.mod versions, and wants only X.Y. I'm trying to track down where this is coming from.
> > Looking in the build logs we see it pulls `gcr.io/oss-fuzz-base/base-builder-go@sha256:c7f01a616db0fc554d98edbd7d49a7177197f77fd4e1371bcb7774d7eac8216d`, which is quite old now:
> > ```shell
> > docker run -it --entrypoint /bin/sh gcr.io/oss-fuzz-base/base-builder-go@sha256:c7f01a616db0fc554d98edbd7d49a7177197f77fd4e1371bcb7774d7eac8216d
> > /root/.go/bin/go version
> > go version go1.19 linux/amd64
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > There is a newer image for this which bundles with go1.22:
> > ```shell
> > docker run -it --entrypoint /bin/sh gcr.io/oss-fuzz-base/base-builder-go@sha256:37d560af1d31ae8d339902acaedb8eb6fdf3f04469706f39e3bbea44fa3fdeec
> > /root/.go/bin/go version
> > go version go1.22.1 linux/amd64
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If we can get this to pull the later container for `base-builder-go` then I expect this will resolve itself.
> 
> Does the following changes in `go.mod` work before oss-fuzz updates the container image?
> 
> ```
> go 1.22
> toolchain 1.22.7
> ```

Oh my bad, the container base image is defined in our repo.

**Review by roger2hk — APPROVED** [review]

**Review by roger2hk — APPROVED** [review]

**Pull request merged**